software-modules/langs/java/javac/java.sh

```diff
@@ -36,8 +35,50 @@ savechanges /tmp/$NAME.hsm
 ## Clean package to maintain only relevant files
 hsm2dir /tmp/$NAME.hsm
 cd /tmp/$NAME.hsm
-find . ! -path "./usr/lib/jvm*" ! -path "./usr/bin*" ! -path "./usr/share/icons*" ! -path "./usr/share/man*" ! -path "./usr/share/pixmaps*" ! -path "./etc/alternatives*" ! -path "./usr/share/application-registry*" ! -path "./usr/share/lintian*" ! -path "./usr/share/applications/java-documentation.desktop" ! -path "./usr/share/doc/reference/java*" -delete
+
+# Array to store paths to exclude that contain java things
+declare -a EXCLUDE_PATHS=(
+    "./etc/.java/*"
+    "./etc/alternatives/*"
+    "./etc/java-11-openjdk/*"
+    "./etc/java-17-openjdk/*"
+    "./etc/ssl/*"
+    "./usr/bin/*"
+    "./usr/lib/*"
+    "./usr/lib/jvm/*"
+    "./usr/sbin/update-java-alternatives*"
+    "./usr/share/applications/java-documentation.desktop*"
+    "./usr/share/bash-completion/completions/update-java-alternatives*"
+    "./usr/share/ca-certificates-java/*"
+    "./usr/share/doc/*"
+    "./usr/share/gdb/auto-load/usr/lib/jvm/*"
+    "./usr/share/gtksourceview-4/language-specs/*"
+    "./usr/share/java/*"
+    "./usr/share/lintian/*"
+    "./usr/share/man/man1/*"
+    "./usr/share/man/man8/*"
+    "./usr/share/maven-repo/*"
+    "./usr/share/mime/*"
+    "./usr/share/nano/*"
+    "./usr/share/netbeans/platform18/*"
+    "./usr/share/nvim/runtime/*"
+    "./usr/share/source-highlight/*"
+)
+
+# Construct the find command with the paths to exclude
+FIND_CMD="find ."
+for PATH in "${EXCLUDE_PATHS[@]}"; do
```

Comment:
*Ah, sure...*, you cannot use the variable name `PATH`, it is reserved for the system. Let's remember some examples:
1. `$PATH`: Specifies the directories where the system looks for executable programs.
2. `$HOME`: Points to the home directory of the current user.
3. `$LANG`: Defines the default language for the user interface and locale settings.
4. `$TERM`: Specifies the terminal type being used.
5. `$SHELL`: Identifies the default shell for the user.
6. `$PWD`: Holds the current working directory.

You can check the specifics for Debian [here](https://wiki.debian.org/EnvironmentVariables).

Reply:
Ok, after testing I've been able to make this PR work by changing the name of the PATH variable. Please change it. 
On the other hand, the warnings still appearing, I get it is expected, but I cannot see why this version it's better than having the full list of `rm`. Having those warning could be triggering for someone building the system for the first time, also building the string makes this script way more complicated to understand, and trying to delete the warning is way more complicated than just having the `rm`.

Reply:
I fixed the PATH variable, totally forgot that it was a special variable 😅 .

On the other hand i also improved the find command so it now removes unwanted files in two steps
- Remove every unwanted file
- Remove every empty folder (that should be a leftover of the unwanted files)

And with that in place, now the script executes without a warning and without having to pass a true next to the command so it does not stop execution :D

Reply:
> I fixed the PATH variable, totally forgot that it was a special variable 😅 .
> 
> On the other hand i also improved the find command so it now removes unwanted files in two steps
> 
> * Remove every unwanted file
> * Remove every empty folder (that should be a leftover of the unwanted files)
> 
> And with that in place, now the script executes without a warning and without having to pass a true next to the command so it does not stop execution :D

Sorry, I tested this and I had to add the `|| true`, otherwise it stoped because of the `set -xe`. Also, are you testing this on huronOS? I still not getting why aren't you catching this errors on your side.

Reply:
Yeah, I tested it on Huron, and I don't see why it should require a `|| true` since everything that is being done is expected to not throw warnings 🤔

Reply:
> Yeah, I tested it on Huron, and I don't see why it should require a `|| true` since everything that is being done is expected to not throw warnings 🤔

Would you like me to make ISO de build and sent it to you? This way you can have the same environment as me. Maybe our builds are differnt¿

Reply:
Sure, but before that could you also show me the line before the error? 
the one including the `find . yadayada`

I want to make sure of something before having you making an iso

Reply:
I tried it in a fresh b27 iso install and I can confirm that it works correctly for me 🤔

Reply:
@equetzal 
Any news on this one? 🤔

Reply:
> @equetzal Any news on this one? 🤔

Not sure of what was wrong, but I've been able to run it without any issues.